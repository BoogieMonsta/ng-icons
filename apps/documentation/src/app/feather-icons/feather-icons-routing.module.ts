import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { FeatherIconsComponent } from './feather-icons.component';

const routes: Routes = [{ path: '', component: FeatherIconsComponent }];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class FeatherIconsRoutingModule { }
