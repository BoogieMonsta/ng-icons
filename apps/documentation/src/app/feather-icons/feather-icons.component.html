<div class="prose max-w-full mb-8">
  <h3>Feather Icons</h3>

  <p>
    Install the <code>@ng-icons/feather-icons</code> package to use the
    following icons within your application.
  </p>
</div>

<app-icon-card-list>
  <app-icon-card
    *ngFor="let icon of icons | keyvalue"
    [icon]="icon.key"
  ></app-icon-card>
</app-icon-card-list>
