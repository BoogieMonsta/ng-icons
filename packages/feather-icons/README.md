# feather-icons

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test feather-icons` to execute the unit tests.
