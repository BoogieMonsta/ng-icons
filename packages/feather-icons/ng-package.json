{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/packages/feather-icon",
  "lib": {
    "entryFile": "src/index.ts"
  }
}
