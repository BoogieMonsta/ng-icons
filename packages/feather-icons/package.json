{
  "name": "@ng-icons/feather-icons",
  "version": "0.0.1",
  "repository": {
    "url": "https://github.com/ashley-hunter/ng-icons"
  },
  "homepage": "https://ashley-hunter.github.io/ng-icons/",
  "license": "MIT",
  "peerDependencies": {
    "@angular/common": ">=11.0.0",
    "@angular/core": ">=11.0.0"
  },
  "dependencies": {
    "tslib": "^2.2.0"
  }
}
