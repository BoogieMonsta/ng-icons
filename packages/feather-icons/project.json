{
  "projectType": "library",
  "root": "packages/feather-icons",
  "sourceRoot": "packages/feather-icons/src",
  "prefix": "ng-icons",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:package",
      "outputs": ["dist/packages/feather-icons"],
      "options": {
        "project": "packages/feather-icons/ng-package.json"
      },
      "configurations": {
        "production": {
          "tsConfig": "packages/feather-icons/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "packages/feather-icons/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/feather-icons"],
      "options": {
        "jestConfig": "packages/feather-icons/jest.config.js",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "packages/feather-icons/src/**/*.ts",
          "packages/feather-icons/src/**/*.html"
        ]
      }
    },
    "deploy": {
      "builder": "ngx-deploy-npm:deploy",
      "options": {
        "access": "public"
      }
    }
  },
  "tags": []
}
