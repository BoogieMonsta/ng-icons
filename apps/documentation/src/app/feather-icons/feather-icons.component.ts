import { Component } from '@angular/core';
import * as featherIcons from '@ng-icons/feather-icons';

@Component({
  selector: 'app-feather-icons',
  templateUrl: './feather-icons.component.html',
})
export class FeatherIconsComponent {
  iconset = featherIcons;
}
