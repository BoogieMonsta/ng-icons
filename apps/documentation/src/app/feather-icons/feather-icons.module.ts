import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as featherIcons from '@ng-icons/feather-icons';
import { FeatherIconsRoutingModule } from './feather-icons-routing.module';
import { FeatherIconsComponent } from './feather-icons.component';
import { NgIconsModule } from '@ng-icons/core';
import { SharedModule } from '../shared/shared.module';

@NgModule({
  declarations: [FeatherIconsComponent],
  imports: [
    CommonModule,
    FeatherIconsRoutingModule,
    NgIconsModule.withIcons(featherIcons),
    SharedModule,
  ],
})
export class FeatherIconsModule {}
