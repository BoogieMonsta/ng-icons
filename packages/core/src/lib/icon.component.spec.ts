import { IconComponent, NgIconsModule } from '@ng-icons/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FeatherAlertCircle } from '@ng-icons/feather-icons';

describe('IconComponent', () => {
  let component: IconComponent;
  let fixture: ComponentFixture<IconComponent>;
  let nativeElement: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NgIconsModule.withIcons({ FeatherAlertCircle })],
      declarations: [IconComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(IconComponent);
    component = fixture.componentInstance;
    component.name = 'feather-alert-circle';
    fixture.detectChanges();
    nativeElement = fixture.nativeElement;
  });

  it('should insert the expected template', () => {
    expect(nativeElement).toMatchSnapshot();
  });
});
