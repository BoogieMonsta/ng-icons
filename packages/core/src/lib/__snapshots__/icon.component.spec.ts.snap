// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent should insert the expected template 1`] = `
<div
  id="root0"
  style="--ng-icon__size: 1em;"
>
  <svg
    class="feather feather-alert-circle"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    style="width:var(--ng-icon__size, 1em);height:var(--ng-icon__size, 1em);stroke-width:var(--ng-icon__stroke-width, 2)"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
</div>
`;
